backend/src/routes/api/status/adminAllowedUsers.ts

```diff
@@ -44,11 +43,14 @@ const getUserActivityFromNotebook = async (
 
   return notebooks.items
     .map<[string | undefined, string | undefined]>((notebook) => [
-      notebook.metadata?.annotations?.['opendatahub.io/username'],
-      notebook.metadata?.annotations?.['notebooks.kubeflow.org/last-activity'] ||
-        notebook.metadata?.annotations?.['kubeflow-resource-stopped'],
+      notebook.metadata.annotations['opendatahub.io/username'],
+      notebook.metadata.annotations['notebooks.kubeflow.org/last-activity'] ||
+        notebook.metadata.annotations['kubeflow-resource-stopped'],
```

Comment:
I think I accounted for this with the `string | undefined` type? I'm not sure what else should be changed.

Reply:
It's this definition that's incorrect because `annotations` is defined as an object and not possibly undefined here. What happens if we change this to `annotations?: Partial<{`. Is there a large cascading effect? 
```
export type Notebook = K8sResourceCommon & {
  metadata: {
    annotations: Partial<{
      'kubeflow-resource-stopped': string; // datestamp of stop (if omitted, it is running)
      'notebooks.kubeflow.org/last-activity': string; // datestamp of last use
      'opendatahub.io/username': string; // the untranslated username behind the notebook

      // TODO: Can we get this from the data in the Notebook??
      'notebooks.opendatahub.io/last-image-selection': string; // the last image they selected
      'notebooks.opendatahub.io/last-size-selection': string; // the last notebook size they selected
    }>;
```

Reply:
On first glance there doesn't seem like there is a cascading effect

Reply:
Careful using labels and annotations -- I caught this at random, so forgive if this is inaccurate. I do not believe we can ever assume Notebooks have the annotation or label blocks. If our types are wrong, we should fix those.

Reply:
Oh, didn't read the thread -- Christian caught this 🙂